SimTKcommon/src/Pathname.cpp

```diff
@@ -238,6 +240,161 @@ void Pathname::deconstructPathname( const string&   name,
+    removeDriveInPlace(pathcleaned, pathdrive);
+
+    String swdcleaned = String::trimWhiteSpace(swd).replaceAllChar('\\', '/');
+    if (!swdcleaned.empty() && swdcleaned[swdcleaned.size() - 1] != '/')
+        swdcleaned += '/';
+    removeDriveInPlace(swdcleaned, swddrive);
+
+    // If swd is empty (or only white space) then use deconstructPathname().
+    if (swdcleaned.empty() && swddrive.empty()) {
+        bool dontApplySearchPath;
+        deconstructPathname(path, dontApplySearchPath, directory, fileName, extension);
+        if (!dontApplySearchPath)
+            directory = getCurrentWorkingDirectory() + directory;
+
+    // swd was not empty. If necessary, preprocess the swd. Then concatenate swd and path.
+    // Finally, resolve drives.
+    }
+    else {
+        /* PREPROCESSING THE SWD. */
+        // Preprocess the swd if it leads with a "./"
+        if (swdcleaned.substr(0, 2) == "./") {
+            swdcleaned.replace(0, 2, getCurrentWorkingDirectory(swddrive));
+            removeDriveInPlace(swdcleaned, swddrive);
+            swdcleaned.insert(0, "/"); // ensure swd starts with "/" to denote full path
+        }
+        // Also preprocess the swd if it leads with "/". Just grab current drive letter.
+        else if (swdcleaned.substr(0, 1) == "/" && swddrive.empty()) {
+            swddrive = getCurrentDriveLetter();
+        }
+        // Also preprocess if swd starts with something of the form "C:folder/". Resolve
+        // by finding the current directory of this drive.
+        else if (!swddrive.empty()) {
+            swdcleaned.insert(0, getCurrentWorkingDirectory(swddrive));
+            removeDriveInPlace(swdcleaned, swddrive);
+            swdcleaned.insert(0, "/"); // ensure swd starts with "/" to denote full path
+        }
+
+        /* CHECKING IF THE SWD SHOULD BE PREPENDED TO THE PATH. */
+        // If path was an absolute path (on Windows this includes a drive or begins with "@/")
+        // then we should not prepend the swd.
+        if (pathcleaned.substr(0, 1) == "/" || pathcleaned.substr(0, 2) == "@/") {
+            processed = pathcleaned;
+            // Ensure that if there was a pathdrive, ignore the swddrive.
+            if (!pathdrive.empty()) swddrive = "";
+        }
+
+        // If path starts with "./", we remove the "./" and concatenate with swdcleaned.
+        else if (pathcleaned.substr(0, 2) == "./") {
+            pathcleaned.erase(0, 2);
+            processed = swdcleaned + pathcleaned;
+        }
+        // Looks like a relative path name (i.e. pathcleaned starts immediately with
+        // a directory or file name).
+        else {
+            processed = swdcleaned + pathcleaned;
+        }
+
+        // If the pathname in its entirety is just one of these, append 
+        // a slash to avoid special cases below.
+        if (processed == "." || processed == ".." || processed == "@")
+            processed += "/";
```

Comment:
Looks fine to me since the other way is swd="" and path=".", which would be sent to deconstructPathname().